*** ./src/main/java/org/apache/hadoop/hbase/master/HMaster.java	2010-11-18 09:24:58.000000000 +0000
--- ./src/main/java/org/apache/hadoop/hbase/master/HMaster.java	2010-11-18 09:26:07.000000000 +0000
***************
*** 356,363 ****
    private static String getMyAddress(final Configuration c)
    throws UnknownHostException {
      // Find out our address up in DNS.
!     String s = DNS.getDefaultHost(c.get("hbase.master.dns.interface","default"),
!       c.get("hbase.master.dns.nameserver","default"));
      s += ":" + c.get(HConstants.MASTER_PORT,
          Integer.toString(HConstants.DEFAULT_MASTER_PORT));
      return s;
--- 356,362 ----
    private static String getMyAddress(final Configuration c)
    throws UnknownHostException {
      // Find out our address up in DNS.
!     String s = DNS.getDefaultIP(c.get("hbase.master.dns.interface","default"));
      s += ":" + c.get(HConstants.MASTER_PORT,
          Integer.toString(HConstants.DEFAULT_MASTER_PORT));
      return s;
*** ./src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java.	2010-11-18 09:27:13.000000000 +0000
--- ./src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java	2010-11-18 09:28:20.000000000 +0000
***************
*** 251,259 ****
     * @throws IOException
     */
    public HRegionServer(Configuration conf) throws IOException {
!     machineName = DNS.getDefaultHost(
!         conf.get("hbase.regionserver.dns.interface","default"),
!         conf.get("hbase.regionserver.dns.nameserver","default"));
      String addressStr = machineName + ":" +
        conf.get(HConstants.REGIONSERVER_PORT,
            Integer.toString(HConstants.DEFAULT_REGIONSERVER_PORT));
--- 251,258 ----
     * @throws IOException
     */
    public HRegionServer(Configuration conf) throws IOException {
!     machineName = DNS.getDefaultIP(
!         conf.get("hbase.regionserver.dns.interface","default"));
      String addressStr = machineName + ":" +
        conf.get(HConstants.REGIONSERVER_PORT,
            Integer.toString(HConstants.DEFAULT_REGIONSERVER_PORT));
