Index: jsonrpc/proxy.py
===================================================================
--- jsonrpc/proxy.py	(revision 19)
+++ jsonrpc/proxy.py	(working copy)
@@ -41,7 +41,7 @@
          postdata = dumps({"method": self.__serviceName, 'params': args, 'id':'jsonrpc'})
          respdata = urllib.urlopen(self.__serviceURL, postdata).read()
          resp = loads(respdata)
-         if resp['error'] != None:
+         if resp.has_key('error'):
              raise JSONRPCException(resp['error'])
          else:
              return resp['result']
