*** bin/scalarisctl.in.org	2010-03-09 13:17:37.000000000 -0800
--- bin/scalarisctl.in	2010-03-09 13:18:07.000000000 -0800
***************
*** 110,119 ****
  }
  
  FQDN_SET=`hostname -f | grep "\.*\." | wc -l`
! if [ "x0" = "x$FQDN_SET" ]; then
!     echo "you have to set a fully qualified domain name (FQDN)"
!     exit -1
! fi
  
  case $1 in
      boot)
--- 110,119 ----
  }
  
  FQDN_SET=`hostname -f | grep "\.*\." | wc -l`
! #if [ "x0" = "x$FQDN_SET" ]; then
! #    echo "you have to set a fully qualified domain name (FQDN)"
! #    exit -1
! #fi
  
  case $1 in
      boot)
