*** pom.xml.org	2010-11-18 09:00:00.000000000 +0000
--- pom.xml	2010-11-18 09:01:29.000000000 +0000
***************
*** 477,483 ****
      <compileSource>1.6</compileSource>
      <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
      <hbase.version>0.89.20100924</hbase.version>
!     <hadoop.version>0.20.3-append-r964955-1240</hadoop.version>
  
      <commons-cli.version>1.2</commons-cli.version>
      <commons-logging.version>1.1.1</commons-logging.version>
--- 477,483 ----
      <compileSource>1.6</compileSource>
      <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
      <hbase.version>0.89.20100924</hbase.version>
!     <hadoop.version>0.20.2</hadoop.version>
  
      <commons-cli.version>1.2</commons-cli.version>
      <commons-logging.version>1.1.1</commons-logging.version>
***************
*** 544,549 ****
--- 544,551 ----
          <groupId>org.apache.hadoop</groupId>
          <artifactId>hadoop-core</artifactId>
          <version>${hadoop.version}</version>
+         <scope>system</scope>
+         <systemPath>${basedir}/../lib/hadoop-0.20.2.jar</systemPath>
        </dependency>
        <dependency>
          <groupId>${project.groupId}</groupId>
